server/routes/updateTodo.js

```diff
@@ -3,6 +3,20 @@ const { connect } = require('hyper-connect')
 const hyper = connect(process.env.HYPER)
 
 module.exports = async function updateTodo (req, res) {
-  const result = await hyper.data.update(req.params.id, req.body)
-  res.json(result)
+  try {
+    const { id } = req.params
+    const task = await hyper.data.get(id)
+    if (!task) {
+      res.status(404).json({ message: 'Task does not exist' })
+    } else {
+      if (!req.body.task) {
+        res.status(400).json({ message: 'Task needs to contain text' })
+      } else {
+        const result = await hyper.data.update(id, req.body)
```

Comment:
You will want validation to ensure body is the correct shape. See my comment on #42 . It can be done in a separate PR though

Reply:
Ok